### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by vztot**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                PreparedStatement insertStatement =
+                        connection.prepareStatement(query);
+                insertStatement.setLong(1, order.getOrderId());
+                insertStatement.setLong(2, product.getProductId());
+                insertStatement.executeUpdate();
+            }
+        }
+    }
+
+    private Order createOrderFromResultSetAndOrdersProducts(ResultSet resultSet)
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java]:
```suggestion
    private Order getOrderFromResultSet(ResultSet resultSet)
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            throws SQLException {
+        Long orderId = resultSet.getLong("order_id");
+        Long userId = resultSet.getLong("user_id");
+        Order order = new Order(orderId, createProductListFromOrdersProducts(orderId), userId);
+        return order;
+    }
+
+    private List<Product> createProductListFromOrdersProducts(Long orderId) throws SQLException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java]:
```suggestion
    private List<Product> getOrderProducts(Long orderId) throws SQLException {
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                String name = resultSet.getString("name");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                Product product = new Product(id, name, price);
+                list.add(product);
+            }
+            return list;
+        }
+    }
+
+    private void deleteOrderFromOrdersProducts(Long orderId) throws SQLException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java]:
```suggestion
    private void deleteOrderProducts(Long orderId) throws SQLException {
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        Long id = resultSet.getLong("user_id");
+        String name = resultSet.getString("name");
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
+        User user = new User(id, name, login, password);
+        user.setRoles(createRolesFromUserId(id));
+        return user;
+    }
+
+    private Set<Role> createRolesFromUserId(Long userId) throws SQLException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java]:
```suggestion
    private Set<Role> getUserRoles(Long userId) throws SQLException {
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+                PreparedStatement insertStatement =
+                        connection.prepareStatement(insertQuery);
+                insertStatement.setLong(1, user.getUserId());
+                insertStatement.setLong(2, resultSet.getLong("role_id"));
+                insertStatement.executeUpdate();
+            }
+        }
+    }
+
+    private User createUserFromResultSetAndSetRoles(ResultSet resultSet) throws SQLException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java]:
```suggestion
    private User getUserFromResultSet(ResultSet resultSet) throws SQLException {
```
setting roles it's part of fetching user, so we don't have to ad this to method name

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job!

**Comment by 6ALLIKA** [comment]
5051

**Pull request merged**